zarr/storage.py

```diff
@@ -2224,27 +2224,25 @@ def __iter__(self):
             list_blobs_prefix = self.prefix + '/'
         else:
             list_blobs_prefix = None
-        for blob in self.client.list_blobs(self.container, list_blobs_prefix):
+        for blob in self.client.list_blobs(list_blobs_prefix):
             yield self._strip_prefix_from_path(blob.name, self.prefix)
 
     def __len__(self):
         return len(self.keys())
 
     def __contains__(self, key):
         blob_name = self._append_path_to_prefix(key)
-        if self.client.exists(self.container, blob_name):
-            return True
-        else:
-            return False
+        return self.client.get_blob_client(blob_name).exists()
 
     def listdir(self, path=None):
-        from azure.storage.blob import Blob
+        # from azure.storage.blob import Blob
         dir_path = normalize_storage_path(self._append_path_to_prefix(path))
         if dir_path:
             dir_path += '/'
         items = list()
-        for blob in self.client.list_blobs(self.container, prefix=dir_path, delimiter='/'):
-            if type(blob) == Blob:
+        for blob in self.client.list_blobs(name_starts_with=dir_path):
+            # items.append(self._strip_prefix_from_path(blob.name, dir_path))
+            if '/' not in blob.name:  # what is this doing?
```

Comment:
When the list_blobs will be replaced with walk_blobs with a `delimiter='/'` parameter, the return items will be the contents of the directory. So if the directory structure is like so:
```
.zgroup
foo/.zgroup
foo/bar/.zarray
```
then `walk_blobs(self.container, name_starts_with='foo/', delimiter='/')` will return `['foo/.zgroup', 'foo/bar/']`
The `.zgroup` will be a case of the first condition `if '/' not in blob.name`, which if true, means that we have a blob, otherwise it's a directory. And the code in the else block removes the '/' from the directory.
So return value of `listdir('foo')` should be `['.zgroup', 'bar']`.

I admit, it is not immediately obvious. Any refactoring would be greatly appreciated.